packages/web3torrent/src/library/web3torrent-lib.ts

```diff
@@ -105,11 +105,14 @@ export default class WebTorrentPaidStreamingClient extends WebTorrent {
     return torrent;
   }
 
-  async cancel(torrentInfoHash: string, callback?: (err: Error | string) => void) {
+  async cancel(infoHash: string, callback?: (err: Error | string) => void) {
     log.info('> Peer cancels download. Pausing torrents');
-    const torrent = this.torrents.find(t => t.infoHash === torrentInfoHash);
+    const torrent = this.torrents.find(t => t.infoHash === infoHash);
     if (torrent) {
-      torrent.pause();
+      await this.closeDownloadingChannels(torrent);
+      this.remove(infoHash, callback);
+    } else {
+      callback(new Error('No torrent found'));
```

Comment:
Is the expectation that the callback should handle this error? I note that the error is not thrown.

Reply:
https://github.com/statechannels/monorepo/pull/1533#discussion_r421271946